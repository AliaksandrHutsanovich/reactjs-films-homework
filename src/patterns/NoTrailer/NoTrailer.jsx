import React, { memo } from 'react';

import '!style-loader!css-loader!sass-loader!antd/dist/antd.css'; //eslint-disable-line

import { Text } from '../../components';

import styles from './NoTrailer.scss';

const NoTrailer = () => (
  <div className={styles.wrapper}>
    <Text type="white" size={20}>No Trailer</Text>
  </div>
);

export default memo(NoTrailer);
