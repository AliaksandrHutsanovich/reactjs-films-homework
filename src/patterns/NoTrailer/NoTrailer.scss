.wrapper {
  background-color: rgba(12, 12, 11, 0.4);
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 300px;
  width: 100%;
}
