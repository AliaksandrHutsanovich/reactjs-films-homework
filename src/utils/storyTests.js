const storyTests = (stories, renderer) => {
  Object.keys(stories).forEach((storyKey) => {
    if (typeof stories[storyKey] !== 'function') {
      return;
    }
    it(`${storyKey} should be rendered properly`, () => {
      renderer.render(stories[storyKey]());
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
};

export default storyTests;
