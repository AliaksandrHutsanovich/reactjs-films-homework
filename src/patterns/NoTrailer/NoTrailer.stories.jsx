import React from 'react';
import NoTrailer from './NoTrailer';

export default {
  component: NoTrailer,
  title: 'patterns/NoTrailer',
  decorators: [
    (story) => (
      <div
        style={{
          width: '300px',
          height: '300px',
          padding: '10px',
        }}
      >
        {story()}
      </div>
    ),
  ],
};

export const noTrailer = () => (
  <NoTrailer />
);
