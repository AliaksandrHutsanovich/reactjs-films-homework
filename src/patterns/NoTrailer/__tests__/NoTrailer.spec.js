import ShallowRenderer from 'react-test-renderer/shallow';
import storyTests from '../../../utils/storyTests';
import * as stories from '../NoTrailer.stories';

describe('<NoTrailer />', () => {
  const renderer = new ShallowRenderer();

  storyTests(stories, renderer);
});
