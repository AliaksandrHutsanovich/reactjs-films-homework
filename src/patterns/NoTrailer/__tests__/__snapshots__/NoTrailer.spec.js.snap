// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoTrailer /> noTrailer should be rendered properly 1`] = `
<div
  className="wrapper"
>
  <UNDEFINED
    size={20}
    type="white"
  >
    No Trailer
  </UNDEFINED>
</div>
`;
